scripts/update_snapshots.sh

```diff
@@ -6,16 +6,33 @@ set -eu
 toplevel="$(git -C "$(dirname "$0")" rev-parse --show-toplevel)"
 cd "${toplevel}"
 
-yarn backend
-echo "Updating GitHub graphql flow types"
-node ./bin/generateGithubGraphqlFlowTypes.js > src/plugins/github/graphqlTypes.js
-echo "Updating for sharness/test_load_example_github.t"
+tmpdir="$(mktemp -d)"
+cleanup() {
+    rm -r "${tmpdir}"
+}
+trap cleanup EXIT
+
+SOURCECRED_BIN="${tmpdir}/bin"
+yarn run --silent backend --output-path "${SOURCECRED_BIN}"
+export SOURCECRED_BIN  # for Sharness and shell tests
```

Comment:
Why not just export SOURCECRED_BIN when declaring it?

Reply:
The presence of `SOURCECRED_BIN` generally means “this directory
contains built SourceCred binaries”, but that’s not actually true until
the previous line has run. It won’t matter unless `yarn backend` itself
depends on `SOURCECRED_BIN` (which would be very bad).

Separately, you generally want to separate assignments and exports when
the rhs of the assignment may fail, because `export` will mask that
status: `FOO="$(false)"` has exit status 1 because `false` fails, but
`export FOO="$(false)"` has exit status 0 because the `export` itself
succeeds. This doesn’t matter here, because if `${tmpdir}` fails to
expand then the `export` statement will fail. (Shellcheck will tell you
if you’re doing this wrong.)

Would you like me to move this up?

Reply:
Slight preference for moving it up, since IMO it's a little bit confusing, and it seems the considerations that motivate it don't apply in this particular case. but overall I'm fine either way.